src/NServiceBus.Core/Persistence/InMemory/SubscriptionStorage/InMemorySubscriptionStorage.cs

```diff
@@ -11,21 +11,21 @@ namespace NServiceBus.InMemory.SubscriptionStorage
     /// </summary>
     class InMemorySubscriptionStorage : ISubscriptionStorage
     {
-        void ISubscriptionStorage.Subscribe(Address address, IEnumerable<MessageType> messageTypes)
+        public void Subscribe(string address, IEnumerable<MessageType> messageTypes)
         {
             foreach (var m in messageTypes)
             {
-                var dict = storage.GetOrAdd(m, type => new ConcurrentDictionary<Address, object>());
+                var dict = storage.GetOrAdd(m, type => new ConcurrentDictionary<string, object>());
```

Comment:
I think we need to pass `StringComparer.OrdinalIgnoreCase` to the `ConcurrentDictionary` ctor, because before `Address` did a case insensitive equality, @SimonCropp  ?